source/Calamari.CloudAccounts/AzureAccountFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Calamari.Common.Plumbing.Variables;
+
+namespace Calamari.CloudAccounts
+{
+    public static class AzureAccountFactory
+    {
+        public static IAzureAccount Create(IVariables variables)
+            => !string.IsNullOrEmpty(variables.Get(AzureAccountVariables.OpenIDJwt)) //if we have an OpenIdConnect JWT, return an AzureOidcAccount
+                ? (IAzureAccount)new AzureOidcAccount(variables)
+                : new AzureServicePrincipalAccount(variables);
```

Comment:
We were doing this check-JWT logic in a few places, so consolidated here under a single factory